airflow/providers/clickhouse/hooks/clickhouse.py

```diff
@@ -0,0 +1,139 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""This module allows connecting to a ClickHouse."""
+from typing import Any, Dict, Optional, Tuple
+
+from clickhouse_driver import Client as ClickHouseClient
+
+from airflow import AirflowException
+from airflow.hooks.base import BaseHook
+
+
+class ClickHouseHook(BaseHook):
+    """
+    Interact with ClickHouse.
+    Performs a connection to ClickHouse and retrieves client.
+
+    :param clickhouse_conn_id: Reference to
+        :ref:`ClickHouse connection id<howto/connection:clickhouse>`.
+    :param database: database for the hook, if not provided schema from Connection will be used (optional).
+    """
+
+    conn_name_attr = 'clickhouse_conn_id'
+    default_conn_name = 'clickhouse_default'
+    conn_type = 'clickhouse'
+    hook_name = 'ClickHouse'
+
+    def __init__(
+        self, clickhouse_conn_id: str = default_conn_name, database: Optional[str] = None, *args, **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.clickhouse_conn_id = clickhouse_conn_id
+        self.database = database
+
+        self.client: Optional[ClickHouseClient] = None
+        self.get_conn()
+
+    def get_conn(self) -> ClickHouseClient:
+        """Function that initiates a new ClickHouse connection"""
+        if self.client is not None:
+            return self.client
+
+        conn = self.get_connection(self.clickhouse_conn_id)
+        connection_kwargs = conn.extra_dejson.copy()
+
+        if conn.port:
+            connection_kwargs.update(port=int(conn.port))
+        if conn.login:
+            connection_kwargs.update(user=conn.login)
+        if conn.password:
+            connection_kwargs.update(password=conn.password)
+
+        #  if  database is provided use it or use from schema
+        if self.database:
+            connection_kwargs.update(database=self.database)
+        elif conn.schema:
+            connection_kwargs.update(database=conn.schema)
```

Comment:
I think this is a source for confusion? Maybe we should customize the connection in the like we do with other providers?

Reply:
Could you please provide more information?

here, if someone wants to override the default schema ( provided by connection ), they can  pass the database argument.